Sources/NIOHTTP2/HTTP2ChannelHandler.swift

```diff
@@ -160,6 +197,10 @@ public final class NIOHTTP2Handler: ChannelDuplexHandler {
         self.writeBuffer = context.channel.allocator.buffer(capacity: 128)
 
         if context.channel.isActive {
+            // handlerAdded is always first, so we can confidently assert our initial state.
+            // We jump immediately to activated here, as channelActive has probably already passed.
+            precondition(self.activationState == .idle)
```

Comment:
I understand the reasoning here but where do we draw the line between using `precondition` and `impossibleActivationStateTransition`?

Reply:
Good question. In this instance my view was that it isn't "network-reachable": this can only be hit if the user actively uses the `NIOHTTP2Handler` twice. However, we can still error here and so we probably should.